### PR TITLE
Add contribution guidelines

### PR DESCRIPTION
**Pull request by aareet**
### Description
Add contribution guidelines. Discuss improvements on this PR. Once this is final, I'll add the same guide to our other repositories.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+- [Pull Request Lifecycle](#pull-request-lifecycle)
+  - [Getting Your Pull Requests Merged Faster](#getting-your-pull-requests-merged-faster)
+
+<!-- /MarkdownTOC -->
+
+## Proposing a Change
+
+In order to be respectful of the time of community contributors, we aim to discuss potential changes in GitHub issues prior to implementation. That will allow us to give design feedback up front and set expectations about the scope of the change, and, for larger changes, how best to approach the work such that the team can review it and merge it along with other concurrent work.
+
+If the bug you wish to fix or enhancement you wish to implement isn't already covered by a GitHub issue that contains feedback from the team, please do start a discussion (either in [a new GitHub issue](https://github.com/terraform-providers/terraform-provider-helm/issues/new/choose) or an existing one, as appropriate) before you invest significant development time. If you mention your intent to implement the change described in your issue, the team can, as best as possible, prioritize including implementation-related feedback in the subsequent discussion.
```
Comment (dak1n1) [inline CONTRIBUTING.md:28]: I've read this part a few times, but I'm having trouble understanding what it means. Could you break it down into simpler terms, or maybe walk me through a scenario?

`If you mention your intent to implement the change described in your issue, the team can, as best as possible, prioritize including implementation-related feedback in the subsequent discussion.`

**Review by dak1n1 — COMMENTED** [review]
This looks great! There was only one part that I had trouble understanding. The rest was clear, helpful for setting expectations, and contained some very useful advice.

**Comment by dak1n1** [comment]
We might also want this guideline to the terraform-k8s repo, since there is some confusion about how to contact maintainers to discuss implementation details. https://github.com/hashicorp/terraform-k8s/pull/35#issuecomment-625994316

**Comment by aareet** [comment]
Thanks Stef - I'm going to close this and recreate with a significant refactor I've done.

**Pull request closed without merging**